MVCDemos/Controllers/ApiControllers/DataGridWebApiController.cs

```diff
@@ -104,6 +105,37 @@ orderby i.CompanyName
 
             return Request.CreateResponse(DataSourceLoader.Load(lookup, loadOptions));
         }
+
+        [HttpPost]
+        public HttpResponseMessage Batch(List<DataChange> changes) {
+            foreach(var change in changes) {
+                var order = new Order();
```

Comment:
new Order is not necessary if change type is not "insert". So move order creation to separate condition.
```
Order order;
if(change.Type == "update" || change.Type == "remove") {
    var key = Convert.ToInt32(change.Key);
    order = _nwind.Orders.First(o => o.OrderID == key);
} else {
    order = new Order();
}
```